src/rdkafka_cgrp.c

```diff
@@ -1086,14 +1090,20 @@ static void rd_kafka_cgrp_handle_JoinGroup (rd_kafka_t *rk,
                                           "JoinGroup failed: %s",
                                           rd_kafka_err2str(ErrorCode));
 
-                if (ErrorCode == RD_KAFKA_RESP_ERR_UNKNOWN_MEMBER_ID)
+                if (ErrorCode == RD_KAFKA_RESP_ERR_FENCED_INSTANCE_ID)
+                        rd_kafka_set_fatal_error(rkcg->rkcg_rk, ErrorCode,
```

Comment:
it might be good if every time you call poll on a client in a failed state, you got a fatal error. is that how it works? I looked quickly and it didn't seem like it, but i might have missed it.

Reply:
I dont think that's desirable since the poll call is also used to trigger things like logs, stats, etc, which the application might want to handle before shutting down.
What it will do though is enqueue a single ERR__FATAL event on the poll queue, which should be sufficient for the application to add fatal error handling, if so desired.